### PR TITLE
OMG-203 Refactor exit validators to use ethereum event listener code

### PR DESCRIPTION
**Pull request by pgebal**

### CODE REVIEW

File: apps/omg_watcher/lib/exit_validator/validator.ex [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      blknum = div(utxo_position, @block_offset)
+      txindex = utxo_position |> rem(@block_offset) |> div(@transaction_offset)
+      oindex = utxo_position - blknum * @block_offset - txindex * @transaction_offset
+      :ok = validate_exit(%{blknum: blknum, txindex: txindex, oindex: oindex}, utxo_exists_callback)
+    end
+
+    :ok
+  end
+
+  defp validate_exit(%{blknum: blknum, txindex: txindex, oindex: oindex} = utxo_exit, utxo_exists_callback) do
```
Comment (pthomalla) [inline apps/omg_watcher/lib/exit_validator/validator.ex]: strange validate exit doing something more than validate.
What you to remove callback?  from this module.

File: apps/omg_api/lib/application.ex [hunk 2]
```diff
@@ -35,12 +35,12 @@ defmodule OMG.API.Application do
       {OMG.API.RootchainCoordinator, MapSet.new([:depositer, :exiter])},
       worker(
         OMG.API.EthereumEventListener,
-        [depositor_config, &OMG.Eth.get_deposits/2, &State.deposit/1],
-        id: :depositor
+        [depositer_config, &OMG.Eth.get_deposits/2, &State.deposit/1, &OMG.Eth.get_root_deployment_height/0],
```
[the remark this answers is not in the document]
Comment (pgebal) [reply apps/omg_api/lib/application.ex]: done

File: apps/omg_api/lib/ethereum_event_listener.ex [hunk 3]
```diff
@@ -21,38 +21,40 @@ defmodule OMG.API.EthereumEventListener do
...
+    {:ok, last_event_block_height} = last_event_block_height_callback.()
 
     height_sync_interval = Application.get_env(:omg_api, :rootchain_height_sync_interval_ms)
     {:ok, _} = schedule_get_events(height_sync_interval)
 
-    _ = Logger.info(fn -> "Starting EthereumEventListener" end)
+    _ = Logger.info(fn -> "Starting EthereumEventListener for #{service_name}" end)
 
-    :ok = RootchainCoordinator.check_in(parent_start_height, service_name)
+    :ok = RootchainCoordinator.check_in(last_event_block_height, service_name)
```
[the remark this answers is not in the document]
Comment (pgebal) [reply apps/omg_api/lib/ethereum_event_listener.ex]: done

File: apps/omg_api/lib/ethereum_event_listener.ex [hunk 4]
```diff
@@ -21,38 +21,40 @@ defmodule OMG.API.EthereumEventListener do
...
-        {%{block_finality_margin: finality_margin, service_name: service_name}, get_ethereum_events_callback,
-         process_events_callback}
+        {%{block_finality_margin: finality_margin, synced_height_update_key: update_key, service_name: service_name},
+         get_ethereum_events_callback, process_events_callback, last_event_block_height_callback}
       ) do
-    # TODO: initialize state with the last ethereum block we have seen events from
-    {:ok, parent_start_height} = Eth.get_root_deployment_height()
+    {:ok, last_event_block_height} = last_event_block_height_callback.()
 
     height_sync_interval = Application.get_env(:omg_api, :rootchain_height_sync_interval_ms)
```
[the remark this answers is not in the document]
Comment (pgebal) [reply apps/omg_api/lib/ethereum_event_listener.ex]: done

File: apps/omg_db/lib/leveldb_server.ex [hunk 5]
```diff
@@ -83,7 +83,9 @@ defmodule OMG.DB.LevelDBServer do
     :last_deposit_block_height,
     :last_fast_exit_block_height,
     :last_slow_exit_block_height,
-    :last_block_getter_synced_height
+    :last_block_getter_synced_height,
+    :last_depositer_block_height,
+    :last_exiter_block_height
```
[the remark this answers is not in the document]
Comment (pgebal) [reply apps/omg_db/lib/leveldb_server.ex]: done

File: apps/omg_watcher/lib/exit_validator/validator.ex (hunk 1 above)
Comment (pgebal) [reply apps/omg_watcher/lib/exit_validator/validator.ex]: done

**Comment by pdobacz** [comment]
@pgebal is updating `docs/architecture.md` (descriptions and interactions between processes in the diagram) necessary here?

**Pull request merged**